client/src/screens/AddItemScreen/AddAndUpdateItemScreen.tsx

```diff
@@ -0,0 +1,193 @@
+import {Image, NativeSyntheticEvent, ScrollView, TextInputChangeEventData, TouchableOpacity, View} from "react-native";
+import React, {useState} from "react";
+import axiosInstance from "../../services/axios.service";
+import {
+    AddItemScreenWrapper,
+    ContentWrapper,
+    ImageListWrapper,
+    SelectedImageWrapper,
+    StyledButton,
+    StyledImage,
+    StyledText,
+    StyledTextInput
+} from "./styles";
+import {StackScreenProps} from "@react-navigation/stack";
+import {HomeStackParamList} from "../../utils/navigation-stack";
+import Toast from "react-native-toast-message";
+import PopUpMenu from "../../components/PopUpMenu";
+import Spinner from 'react-native-loading-spinner-overlay';
+import {selectEmail} from "../../store/user-details/user-details.selectors";
+import {useAppDispatch, useAppSelector} from "../../store/hooks";
+import {editItem} from "../../services/items.service";
+import {ItemWithCategory} from "../../models/item";
+import {getFavorites} from "../../services/favorites.service";
+import {upsertFavoritesThunk} from "../../store/favorites/favorites.slice";
+
+export const defaultColor = '#2196f3';
+type AddItemScreenProps = StackScreenProps<HomeStackParamList, 'AddItem'>;
+
+export const AddAndUpdateItemScreen = ({navigation, route}: AddItemScreenProps) => {
```

Comment:
Call it `UpsertItemScreen`.
Also rename the package and inside `styles.ts`

Reply:
done